rust/oak/src/lib.rs

```diff
@@ -77,12 +83,64 @@ impl ReceiveChannelHalf {
     }
 }
 
-impl Read for ReceiveChannelHalf {
-    fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
-        Ok(unsafe { wasm::channel_read(self.handle, buf.as_mut_ptr(), buf.len()) })
+impl ReceiveChannelHalf {
```

Comment:
I think we should also stop implementing Write for channels, they will only write a message at a time anyways. We may reintroduce something like Read and Write later on on top of special channels that read and write streams of bytes by serializing them over messages, but for now it seems clearer not to mix the two concepts. WDYT?